src/main/java/clueless/Game.java

```diff
@@ -425,14 +425,16 @@ public void loseGame(String charName) {
 	public Player whoHasClue(List<Card> suggestion) {
 		
 		// get next player index to start loop
+		Player currentPlayer = this.getCurrentPlayer();
 		String nextPlayerCharName = this.getNextPlayer().characterName;
 		int index = getCharacterIndexInTurnOrder(nextPlayerCharName);
 		
 		int count = 0;
 		while (count < CHARACTER_TURN_ORDER.length) { // loop over ordered players and get first player who has clue
 
 			if (this.isPlayer(CHARACTER_TURN_ORDER[index]) 
-					&& this.getPlayer(CHARACTER_TURN_ORDER[index]).hasClue(suggestion)) {
+					&& this.getPlayer(CHARACTER_TURN_ORDER[index]).hasClue(suggestion) 
+					&& (this.getPlayer(CHARACTER_TURN_ORDER[index])!= currentPlayer)) {
```

Comment:
```suggestion
					&& (!this.getPlayer(CHARACTER_TURN_ORDER[index]).equals(currentPlayer))) {
```

you should use the .equals() method as this will ensure the condition performs the correct comparison